winchan.js

```diff
@@ -116,8 +116,8 @@ var WinChan = (function() {
         var iframe;
 
         // sanity check, are url and relay_url the same origin?
-        var origin = extractOrigin(opts.url);
-        if (origin !== extractOrigin(opts.relay_url)) {
+        var origin = opts.origin || extractOrigin(opts.url);
+        if (!opts.origin && origin !== extractOrigin(opts.relay_url)) {
```

Comment:
dont pay attention to the text. I am talking about the validation itself. should the relay url have the same host than the origin (worried about the relay iframe)?

about `postMessage`, cross domain messages works fine (as long you set the correct origin).